mypy/main.py

```diff
@@ -259,14 +256,15 @@ def process_options(args: List[str]) -> Tuple[List[BuildSource], Options]:
               file=sys.stderr)
 
     # Check for invalid argument combinations.
-    code_methods = sum(bool(c) for c in [special_opts.modules,
-                                         special_opts.command,
-                                         special_opts.package,
-                                         special_opts.files])
-    if code_methods == 0:
-        parser.error("Missing target module, package, files, or command.")
-    elif code_methods > 1:
-        parser.error("May only specify one of: module, package, files, or command.")
+    if require_targets:
+        code_methods = sum(bool(c) for c in [special_opts.modules,
+                                            special_opts.command,
+                                            special_opts.package,
+                                            special_opts.files])
+        if code_methods == 0:
+            parser.error("Missing target module, package, files, or command.")
+        elif code_methods > 1:
+            parser.error("May only specify one of: module, package, files, or command.")
```

Comment:
Because this diff is pretty long, I'd rather not add unrelated changes to it.  Checking for that isn't completely trivial.